File: paper/rust/bug1776141/src/main.c
#include <stdio.h>

typedef struct msgBuff {
    char message[128];
} TAB_BUFFER;

int main(void) {
    TAB_BUFFER buf;
    char *type = "dummy";
    char *pdbentry = "dummy";

    // Write >128 chars
    char *pfield_name = 
        "123456789012345678901234567890"
        "123456789012345678901234567890"
        "123456789012345678901234567890"
        "123456789012345678901234567890"
        "123456789012345678901234567890";

    sprintf(buf.message, "%-4sL %s %s", pfield_name,
        type, pdbentry
    );

    return 0;
}

File: paper/rust/bug1776141/src/main.rs
struct MsgBuff {
    message: [u8; 128]
}

fn main() {
    let mut m = MsgBuff {
        message: [0; 128]
    };

    let type_ = "dummy";
    let pdbentry = "dummy";
    let pfield_name =
        "123456789012345678901234567890\
         123456789012345678901234567890\
         123456789012345678901234567890\
         123456789012345678901234567890\
         123456789012345678901234567890";

    let formatted = format!("{:<4}L {} {}",
        pfield_name, type_, pdbentry);
    m.message.copy_from_slice(formatted.as_bytes());
}
